### DISCUSSION TITLE
yum install failed to construct an image based on gocd/gocd-agent-centos 9:v23.1.0

### DISCUSSION THREAD

**Discussion by xlw712** [gocd/gocd#11715] [Q&A]
I intend to use gocd/ Gocd-agent-centos 9:v23.1.0 as the base image to package some tools, but encountered a very strange phenomenon, ```docker build``` has the ```yum install ``` command [getaddrinfo() thread failed to start] wrong。
Below is a screenshot of the error
![image](https://github.com/gocd/gocd/assets/10722207/353f7e56-3a28-4437-8f0e-917b1c422969)
I'm fine with gocd/gocd-agent-centos ```7```:v23.1.0。
I have confirmed that dns is OK and I can install it normally when using docker run-itd. Is this a known problem?

**Comment by chadlwilson** [comment]
Did you change to the root user in your Dockerfile before trying to yum install, and then change back to the `go` user afterwards?

**Reply by xlw712** [reply]
I failed when I change to the root user 
![image](https://github.com/gocd/gocd/assets/10722207/c4ed8d5f-abd1-4798-84f5-d174d3f624fb)

**Reply by chadlwilson** [reply]
Ok, don't know then. This is very unlikely to be something to do with gocd's docker image and more likely to do with the way you are trying to build your image, your docker environment or host OS etc. If you're trying to do so cross-architecture (using an x64 docker env to build an arm64 image) there are extra things to do.

Try and build a test child image from `quay.io/centos/centos:stream9` with a `yum install` - if you still have issues then the issue is likely with your host OS environment.

**Reply by xlw712** [reply]
OK,It's strange that the same command works on centos7, but not on centos9. Is there any difference in how the two images are built? Is there a Dockerfile? I want to compare it. I noticed that some commands on Centos9 appear ReadOnly

**Reply by chadlwilson** [reply]
https://github.com/gocd/docker-gocd-agent-centos-7/tree/v23.1.0
https://github.com/gocd/docker-gocd-agent-centos-9/tree/v23.1.0

**Reply by chadlwilson** [reply]
Sometimes newer versions of software or libraries require/use system calls that the container runtime blocks. Centos 7 is very old. Sometimes depends on host OS version, docker version, use of secconp/apparmor etc etc. Google the error message if you want to find other possible reasons.

Or might be something resolved in later Centos Stream 9 builds. You could try off an experimental unreleased image like https://hub.docker.com/gocdexperimental/gocd-server-centos-9 and see if it's any different. The unreleased version works off microdnf rather than dnf which might also be a reason for different behaviour though.
